projects/angular/src/data/datagrid/utils/key-navigation-grid.controller.ts

```diff
@@ -224,7 +225,11 @@ export class KeyNavigationGridController implements OnDestroy {
         y = 0;
       }
     } else if (e.key === Keys.PageUp) {
-      y = y - itemsPerPage > 0 ? y - itemsPerPage + 1 : 1;
+      if (this.strategy === 'virtualScroller') {
+        y = numOfRows / 4 < y ? Math.floor(numOfRows / 4) : 1;
```

Comment:
Why we do division exactly by 4?

Reply:
It's kinda magic number, but the reason is to put the next Index in the upper 1/4 of the total rows.

The calculation is as follows in virtual scroll.
`numOfRows` are all rendered rows. The count is static and don't change, except in some edge cases that don't matter.
Relativley speaking we have not visible 1/4 above, visible 2/4 and invisible 1/4 under.

`PageUp` tries to scroll up a page placing the index at the above 1/4. That will pull down the row and will make the item visible and centered.

This all is relative and `itemsPerPage` also have weight, but I didn't find stable algorithm to implement.
Using `numOfRows / 4` was the most stable and relaiable in my tests.

Reply:
Using `numOfRows / 4` is the most stable and relaiable algorithim.

The most important changes are in the `minBufferPx` and `maxBufferPx`. 
Users have to add significant buffer for virtual scroll to update the rows reliably.

Reply:
```suggestion
        const threshold = numRows / 4;
        y = y >= threshold ? Math.floor(threshold) : 1;
```

I think the `numRows / 4` should be extracted so that it only happens once in the code. I also flipped the condition because I think it reads better that way, but maybe that's just me.

Please also explain the threshold logic (rename if desired) in a code comment so that it is is available to anyone reading the code.

Reply:
After close examination `PageUp` beahviour is actually fixed in #1695.

In this PR is left only a slight Key navigation focus optimization, all other changes are reverted.